src/coreComponents/constitutive/fluid/multifluid/compositional/CompositionalMultiphaseFluid.cpp

```diff
@@ -208,7 +173,71 @@ CompositionalMultiphaseFluid::createKernelWrapper()
                         m_totalDensity.toView() );
 }
 
-REGISTER_CATALOG_ENTRY( ConstitutiveBase, CompositionalMultiphaseFluid, string const &, Group * const )
+// Create the fluid models
+template< typename FLASH, typename PHASE1, typename PHASE2, typename PHASE3 >
+void CompositionalMultiphaseFluid< FLASH, PHASE1, PHASE2, PHASE3 >::createModels()
+{
+  m_componentProperties = std::make_unique< compositional::ComponentProperties >(
+    m_componentCriticalPressure,
+    m_componentCriticalTemperature,
+    m_componentCriticalVolume,
+    m_componentAcentricFactor,
+    m_componentVolumeShift,
+    m_componentBinaryCoeff );
+
+  m_flash = std::make_unique< FLASH >( getName() + '_' + FLASH::catalogName(),
+                                       m_componentNames,
```

Comment:
Ditto.  Should this also be kept with `m_componentProperties`?

Reply:
The `m_componentMolarWeight` and `m_componentNames` members are part of the base `MultiFluidBase` class. So initially I thought it wouldn't be a good idea to lump them in. But having thought about it, I don't see why not.